src/main/java/de/tum/in/www1/artemis/domain/Attachment.java

```diff
@@ -121,19 +138,13 @@ else if (attachmentType == AttachmentType.FILE && link != null && link.contains(
      */
     @PreUpdate
     public void onUpdate() {
-        handleFileChange();
-    }
-
-    private void handleFileChange() {
         if (attachmentType == AttachmentType.FILE && getLecture() != null) {
-            // move file and delete old file if necessary
-            var targetFolder = Path.of(FilePathService.getLectureAttachmentFilePath(), getLecture().getId().toString()).toString();
-            link = fileService.manageFilesForUpdatedFilePath(prevLink, link, targetFolder, getLecture().getId(), true);
+            Path targetFolder = FilePathService.getLectureAttachmentFilePath().resolve(getLecture().getId().toString());
```

Comment:
Consider putting the targetFolder generation in a function (as it avoids code duplication with beforeCreate()).

Reply:
This has different base paths, different IDs and then calls different methods. I'd rather leave it like that. I think it would only become more complicated.